frontend/controller/actions/group.js

```diff
@@ -970,7 +911,36 @@ export default (sbp('sbp/selectors/register', {
   ...encryptedAction('gi.actions/group/sendPaymentThankYou', L('Failed to send a payment thank you note.')),
   ...encryptedAction('gi.actions/group/groupProfileUpdate', L('Failed to update group profile.')),
   ...encryptedAction('gi.actions/group/proposal', L('Failed to create proposal.')),
-  ...encryptedAction('gi.actions/group/proposalVote', L('Failed to vote on proposal.')),
+  ...encryptedAction('gi.actions/group/proposalVote', L('Failed to vote on proposal.'), async (sendMessage, params) => {
+    const state = await sbp('chelonia/contract/state', params.contractID)
+    const data = params.data
+    const proposalHash = data.proposalHash
+    const proposal = state.proposals[proposalHash]
+    const type = proposal.data.proposalType
+    let passPayload
+
+    if (data.vote === VOTE_FOR) {
+      passPayload = {}
+      if (oneVoteToPass(state, proposalHash)) {
+        if (type === PROPOSAL_INVITE_MEMBER) {
+          passPayload = await createInvite({
+            contractID: params.contractID,
+            invitee: proposal.data.proposalData.memberName,
+            creatorID: proposal.creatorID,
+            expires: state.settings.inviteExpiryProposal
+          })
+        }
+      }
+    }
+
+    return sendMessage({
+      ...params,
+      data: {
+        ...data,
+        passPayload
+      }
+    })
+  }),
```

Comment:
Why was this moved here?

Reply:
Because none of this belongs into a `.vue` file. This is something that affects Chelonia contract state and the pass payload should be determined by an action. (Ideally, it'd be determined by a contract, but that's harder to do at the moment because an invite needs to be created)